### PR TITLE
switch to use the karma test runner for viewer (Instead of manually loading a browser page)

### PR DESCRIPTION
**Pull request by caffodian**
Part of #23 , now the tests can be run with "npm test"

### CODE REVIEW

File: viewer/test/test-add-heading-anchors.js [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
+
+  beforeEach(function () {
+    fixture.load("heading-fixtures.html");
+    this.testArea = fixture.el.firstChild;
+    addHeadingAnchors.init("#testarea", "#testarea .popovers");
+  });
+
+
+  afterEach(function () {
+    fixture.cleanup();
```
Comment (caffodian) [inline viewer/test/test-add-heading-anchors.js:18]: side effect of cleaning it up to use fixtures, test ordering doesn't matter anymore

File: viewer/test/test-popovers-hiding.js [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
   before(function () {
-    this.testArea = document.getElementById("test-popovers-hiding");
-    addHeadingAnchors.init("#test-popovers-hiding", "#test-popovers-hiding .popovers");
+    fixture.setBase("fixtures");
+  });
+
+  beforeEach(function () {
+    fixture.load("heading-fixtures.html");
+    this.testArea = fixture.el.firstChild;
+    addHeadingAnchors.init("#testarea", "#testarea .popovers");
```
Comment (bocribbz) [inline viewer/test/test-popovers-hiding.js]: so easy to misread this to `textarea` :)

**Review by bocribbz — APPROVED** [review]

File: viewer/test/test-popovers-hiding.js (hunk 2 above)
Comment (caffodian) [reply viewer/test/test-popovers-hiding.js]: hmm testcontainer might be better then

Comment (bocribbz) [reply viewer/test/test-popovers-hiding.js]: +1 `testcontainer`

**Pull request merged**